### ISSUE TITLE
 Little bug in \insertsubsectionnavigation

### ISSUE THREAD

**Issue by polgab** [josephwright/beamer#424]
Here is the correct implementation for `\insertsubsectionnavigation`:

```
\def\insertsubsectionnavigation#1{%
  \hbox to #1{%
    \vbox{{%
      \usebeamerfont{subsection in head/foot}\usebeamercolor[fg]{subsection in head/foot}%
      \vskip0.5625ex%
      \beamer@currentsubsection=0%
      \def\sectionentry##1##2##3##4##5{}%
      \def\slideentry##1##2##3##4##5##6{\ifnum##6=\c@part\ifnum##1=\c@section%
        \ifnum##2>\beamer@currentsubsection%
        \beamer@currentsubsection=##2%
        \def\insertsubsectionhead{##5}%
        \def\insertsectionheadnumber{##1}%
        \def\insertsubsectionheadnumber{##2}%
        \def\insertpartheadnumber{##6}%
        \setbox\beamer@tempbox=\hbox{\beamer@link(##4){%
            \hbox to #1{\hskip0.3cm%
              \ifnum\c@subsection=##2%
                {\usebeamertemplate{subsection in head/foot}}%
              \else%
                {\usebeamertemplate{subsection in head/foot shaded}}%
              \fi\hfil\hskip0.3cm}}}%
        \ht\beamer@tempbox=1.6875ex\dp\beamer@tempbox=0.75ex%
        \box\beamer@tempbox\fi\fi\fi}%
      \dohead\vskip0.5625ex}}\hfil}}
```

(note the position of `\hfil` into the last line !!!)

**Comment by louisstuart96** [comment]
This was fixed in [83e30fc](https://github.com/josephwright/beamer/commit/83e30fc8640493f2edf68ffd6cbf23cf8bb963fc). 
We will upload the new version to CTAN as soon as possible @josephwright .

**Comment by josephwright** [comment]
@louisstuart96 I'll sort a release this weekend
